### DISCUSSION TITLE
Subclass with subtype attribute and inherited mutator

### DISCUSSION THREAD

**Discussion by andrew222651** [microsoft/pyright#5320] [Q&A] [answered]
Apologies if this is already addressed somewhere. Pyright doesn't find an error in the following. Is this supposed to be allowed by Python's type system?
```python
class Parent:
    a: float

    def __init__(self):
        self.a = 1

    def set(self):
        self.a = 2.5

class Child(Parent):
    a: int
```

**Comment by erictraut** [comment]
Yes, this is allowed in Python's type system. A subclass can override an attribute type of a parent as long as it is a subtype. If you were to redefine `a` as `str` in the subclass and enable the `reportIncompatibleVariableOverride` diagnostic rule in pyright, you'd receive an error.

```python
# pyright: reportIncompatibleVariableOverride=true

class Child(Parent):
    a: str
```

Mypy also does not generate an error for your code sample.

**Reply by erictraut** [reply, marked as answer]
Interestingly, someone posted a similar bug report in the mypy issue tracker just a few minutes before you posted here. https://github.com/python/mypy/issues/15455. Check out that thread for more details.
